scripts/version-linter.py

```diff
@@ -20,11 +20,12 @@ def set_error():
 
 def branch_commit(branch):
   print ('Retrieving', branch, 'head commit...')
-  result=subprocess.run(['git','log','-n','1','--format="%h"','--abbrev=7','--no-merges',f'{branch}'],
+  result=subprocess.run(['git','log','-n','1','--format="%h"','--abbrev=7','--no-merges',f'{branch}','--'],
```

Comment:
what does this change do?

Reply:
@psteckler I'm still testing this. CI fails in `brach_commit` with
```
Retrieving origin/volhovm/13901-pickles-types head commit...
command stdout:
command stderr: fatal: ambiguous argument 'origin/volhovm/13901-pickles-types': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

While on my machine it seems fine:

```
ї> git log -n 1 --format="%f" --abbrev=7 --no-merges origin/volhovm/13901-pickles-types 
13901-Implement-accessors-via-derive_ppx
```

I thought to try to add `--` to the end of the command as recommended in the `stderr` above, and also here: https://stackoverflow.com/questions/54586154/git-diff-fatal-ambiguous-argument-unknown-revision-or-path-not-in-working-tree

Reply:
@psteckler It seems to work, the CI now passes. Does it sound reasonable to you?

Reply:
I've seen this error elsewhere, and it eventually resolves. I'm supposing the reason it works in this PR is not due to the change you made, but for the same reason the error resolves elsewhere. I wonder if a `git fetch` before the `git log` would help.